### ISSUE TITLE
Multiple Duo authentication requests during terraform plan

### ISSUE THREAD

**Issue by hahuang65** [Snowflake-Labs/terraform-provider-snowflake#2194]
**Provider Version**

0.75.0

**Terraform Version**

1.5.7

**Describe the bug**

Using `SNOWFLAKE_USER`, `SNOWFLAKE_PASSWORD`, `SNOWFLAKE_AUTHENTICATOR=UsernamePasswordMFA`, `SNOWFLAKE_CLIENT_STORE_TEMPORARY_CREDENTIAL=1` and `SNOWFLAKE_CLIENT_REQUEST_MFA_TOKEN=1` results in Duo pushes for every single request made by Terraform.

I've also tried setting `"true"` instead of `1` for the boolean options as well.

**Expected behavior**

Should request a single time and cache it for the remainder of `terraform plan` or `terraform run`.

**Code samples and commands**

Please add code examples and commands that were run to cause the problem.

**Additional context**

**Comment by afeld** [comment]
I think https://github.com/Snowflake-Labs/terraform-provider-snowflake/issues/2504#issuecomment-1953583147 may explain why this is happening.

**Comment by JESCHO99** [comment]
@sfc-gh-jcieslak What is the current status for this bug? What I see and testend in version 0.92.0 is that even with the "ALLOW_CLIENT_MFA_CACHING" set to true for the account every terraform plan using the same username and password is sending a duo push which makes it very hard to work good localy.

**Comment by sfc-gh-asawicki** [comment]
Hey @JESCHO99. Which OS you are working on?

**Comment by JESCHO99** [comment]
Hey @sfc-gh-asawicki, we are working on Windows.

**Comment by sfc-gh-asawicki** [comment]
Yes, we will update you on this issue as soon as we make any progress with the caching problems. In the meantime, please inform us if you try any of these workarounds or when you come up with something different (it might be helpful).

**Comment by sfc-gh-asawicki** [comment]
@JESCHO99 are you still experiencing the caching problems on windows? If so, could you share the exact setup and the reproduction steps?

**Comment by philippemiron** [comment]
I've having the same issue with :

```
export SNOWFLAKE_AUTHENTICATOR=EXTERNALBROWSER
export SNOWFLAKE_CLIENT_STORE_TEMPORARY_CREDENTIAL=true
```

`terraform plan/apply` open a handful of browser pages when processing.

**Comment by sfc-gh-jmichalak** [comment]
Hi @philippemiron,

What provider version are you using? Also, what are the values of ALLOW_CLIENT_MFA_CACHING and ALLOW_ID_TOKEN account parameters?

**Comment by sfc-gh-jmichalak** [comment]
Closing due to inactivity. If this issue happens again, please open a new issue, or report this to Snowflake Support, and provide reproduction steps.
